src/GeositeFramework/js/Map.js

```diff
@@ -225,23 +225,33 @@ require(['use!Geosite',
                 if (serviceInfo && service.visible && serviceInfo.pluginObject.showServiceLayersInLegend) {
                     service.visibleLayers.sort(function(a, b) { return a - b; });
                     _.each(service.visibleLayers, function(layerId) {
-                        var layer = _.findWhere(service.layerInfos, {id: layerId});
-                        if (!layer) {
-                            return;
-                        }
-
-                        var serviceLegend = getServiceLegend(service);
-                        if (!serviceLegend) {
-                            return;
-                        }
-
-                        var legend = _.findWhere(serviceLegend, {layerId: layerId});
-                        if (isLayerInScale(service, layer)) {
-                            result.push({
-                                service: service,
-                                layer: layer,
-                                legend: legend
-                            });
+                        var layer,
+                            legend;
+
+                        if (_.isUndefined(service.description) === false) {
+                            if (service.description.match(/WMS/i)) {
```

Comment:
An error occurs when `service.description` is null. Looks like it happens because the legend tries to render before the map service is completely loaded.

Steps:

1. Load WMS layer
2. Load AGS layer (ex. Study Area)